src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java

```diff
@@ -1649,6 +1648,43 @@ else if(getSendParameterValuesAsPostBody()) {
             }
             StringEntity requestEntity = new StringEntity(entityBodyContent.toString(), charset);
             entity.setEntity(requestEntity);
+        } else if (hasArguments()) {
```

Comment:
Is it possible to extract this long code piece into separate method? For better readability.

Reply:
Done